python/ray/serve/_private/proxy.py

```diff
@@ -1032,13 +1036,17 @@ async def send_request_to_replica(
                 async for message in self.timeout_response(proxy_request, request_id):
                     yield message
         except asyncio.CancelledError:
```

Comment:
Thanks for the refactor Alexey, but like what Ed said, I also feel this is not going to catch the cancelled error from the inner response call 😅 will wait for the test to look deeper

Reply:
Not sure i follow your point, Gene. Can you elaborate?

Reply:
oh I guess that cancel error is actually getting raised in the proxy as well. Just thought some test might fail there, but seems like its working fine 😄